docs/testing-deployment-publishing/branching-collaboration/branching.md

```diff
@@ -0,0 +1,233 @@
+- Know more about Merge vs Rebase [**here**](#merge-vs-rebase).
+
+:::
+
+<div style={{
+    position: 'relative',
+    paddingBottom: 'calc(56.67989417989418% + 41px)', // Keeps the aspect ratio and additional padding
+    height: 0,
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/VHuZbhTfT8Q8GbaMzMWA?embed&show_copy_link=true"
+        title="Sharing a Project with a User"
+        style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            colorScheme: 'light'
+        }}
+        frameborder="0"
+        loading="lazy"
+        webkitAllowFullScreen
+        mozAllowFullScreen
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+
+3. Now, you'll see a screen that will display if there are any conflicts. If you don’t have any conflicts, you can simply go ahead and click **Merge Branch**.
+
+![merge](../imgs/merge.avif)
+
+### Resolve merge conflicts
+A merge conflict occurs when multiple team members make changes to the same part of the project such as modify the same screen, widget, or configuration simultaneously.
+
+For example, imagine two developers, Alice and Bob, are working on the same FlutterFlow project and both decide to update the same button widget.
+
+**Alice's Changes (Branch: `feature-alice`)**
+
+- Changes the button text to "Submit Form"
+- Changes the button color to blue
+
+**Bob's Changes (Branch: `feature-bob`)**
+
+- Changes the button text to "Send"
+- Changes the button color to green
+
+When Alice's changes are merged into the main project first, her updates will be integrated without any issues. However, when Bob tries to merge his changes afterward, a merge conflict will occur because the changes to the button text and color have already been modified by Alice.
+
+If you get merge conflicts, here’s how you resolve them.
+
+<div style={{
+    position: 'relative',
+    paddingBottom: 'calc(56.67989417989418% + 41px)', // Keeps the aspect ratio and additional padding
+    height: 0,
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/Ttcvg8zDKtlQSakqppOf?embed&show_copy_link=true"
+        title="Sharing a Project with a User"
+        style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            colorScheme: 'light'
+        }}
+        frameborder="0"
+        loading="lazy"
+        webkitAllowFullScreen
+        mozAllowFullScreen
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+
+For certain conflicts, such as those involving variables and API configurations, you'll see a **View Configuration**     option. Enabling this option allows you to easily identify what has changed.
```

Comment:
Correct me if I'm wrong, but isnt this part about the generated code differences? If yes, i feel it should be highlighted in this section with a bit more details.

Reply:
Yes, internally, it is a generated code difference only, but we don't show this option for widget conflicts (although there is a conflict in the widget's generated code).